game-core/src/test/java/games/strategy/triplea/delegate/battle/MockBattleState.java

```diff
@@ -0,0 +1,34 @@
+package games.strategy.triplea.delegate.battle;
+
+import games.strategy.engine.data.Unit;
+import java.util.Collection;
+import java.util.List;
+import lombok.Builder;
+import lombok.NonNull;
+
+/**
+ * Simple implementation of BattleState for tests to use
+ *
+ * <p>{@link #givenBattleState()} will return a builder with everything defaulted and the test can
+ * override the specific items needed.
+ */
+@Builder
+public class MockBattleState implements BattleState {
+
+  final @NonNull Collection<Unit> attackingUnits;
+  final @NonNull Collection<Unit> defendingUnits;
+
+  public static MockBattleState.MockBattleStateBuilder givenBattleState() {
```

Comment:
It looks like all usages of this method always define attacking unit and defending units, why not just use the builder directly?

IE: instead of: ```MockBattleState.givenBattleState()```, use instead: ```MockBattleState.builder()```

Reply:
Because some of the steps will require more items added.  An example is the aa guns.  They don't care about offensive/defensiveUnits but use the offensive/defensive aa.

I'm actually thinking that all the parameters for `BattleState` is going to end up looking like the parameters to `BattleSteps`.

Reply:
My point is that you don't need to use a builder to init the values to empty lists and then return that builder to overwrite the values. You can use builder directly I would think. Look into using `@Singular` on those: https://projectlombok.org/features/Builder#singular